### ISSUE TITLE
Packer docker only build is failing

### ISSUE THREAD

**Issue by ashafer01** [linkedin/iris#488]
```
[10:07:59] ip-192-168-2-110:~/git/iris/ops/packer 1 % git pull                                                               
Already up to date.
[10:08:07] ip-192-168-2-110:~/git/iris/ops/packer % git status                                                             
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
[10:08:12] ip-192-168-2-110:~/git/iris/ops/packer % git rev-parse HEAD                                                     
f607fe4c5595469766470b899950d97570e8aebe
[10:08:19] ip-192-168-2-110:~/git/iris/ops/packer % python gen_packer_cfg.py ./iris.yaml | tail -n +2 > ./output/iris.json
[10:08:25] ip-192-168-2-110:~/git/iris/ops/packer % packer build -only=docker ./output/iris.json
...
    docker: Collecting python-ldap==2.4.9 (from iris==0.0.14)
    docker:   Downloading https://files.pythonhosted.org/packages/75/f5/344cb326a9ba48ee31d58bb1b685f538c3e73954e08a0b81e7dcf48304e2/python-ldap-2.4.9.tar.gz (133kB)            
    docker:     Complete output from command python setup.py egg_info:                                                                                                           
    docker:     defines: HAVE_SASL HAVE_TLS HAVE_LIBLDAP_R
    docker:     extra_compile_args:
    docker:     extra_objects:
    docker:     include_dirs: /opt/openldap-RE24/include /usr/include/sasl /usr/include
    docker:     library_dirs: /opt/openldap-RE24/lib /usr/lib
    docker:     libs: ldap_r
    docker:     Traceback (most recent call last):
    docker:       File "<string>", line 1, in <module>
    docker:       File "/tmp/pip-install-_v6LcX/python-ldap/setup.py", line 178, in <module>
    docker:         **kwargs
    docker:       File "/home/iris/env/local/lib/python2.7/site-packages/setuptools/__init__.py", line 144, in setup
    docker:         _install_setup_requires(attrs)
    docker:       File "/home/iris/env/local/lib/python2.7/site-packages/setuptools/__init__.py", line 137, in _install_setup_requires
    docker:         dist.parse_config_files(ignore_option_errors=True)
    docker:       File "/home/iris/env/local/lib/python2.7/site-packages/setuptools/dist.py", line 702, in parse_config_files
    docker:         self._parse_config_files(filenames=filenames)
    docker:       File "/home/iris/env/local/lib/python2.7/site-packages/setuptools/dist.py", line 599, in _parse_config_files
    docker:         (parser.read_file if six.PY3 else parser.readfp)(reader)
    docker:       File "/usr/lib/python2.7/ConfigParser.py", line 324, in readfp
    docker:         self._read(fp, filename)
    docker:       File "/usr/lib/python2.7/ConfigParser.py", line 479, in _read
    docker:         line = fp.readline()
    docker:       File "/home/iris/env/lib/python2.7/encodings/ascii.py", line 26, in decode
    docker:         return codecs.ascii_decode(input, self.errors)[0]
    docker:     UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 378: ordinal not in range(128)
    docker:
    docker:     ----------------------------------------
    docker: Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-_v6LcX/python-ldap/
==> docker: Killing the container: 94466af71d7a4ac3b428f47728568bcb5cad842adb61c0c1143fc7e285af2e46
Build 'docker' errored: Script exited with non-zero exit status: 1

==> Some builds didn't complete successfully and had errors:
--> docker: Script exited with non-zero exit status: 1

==> Builds finished but no artifacts were created.
```

Looks like this version of python-ldap is broken? Current version is 3.1.0. Related to #255 ?

I'm able to install `python-ldap==2.4.9` on python 2.7.10 without issue though, so unsure what source of the problem is.

**Comment by avanier** [comment]
I've been digging through the git history, and I can't see a change to the python-ldap version since January 2017, so it's probably not that. My guess is on a change in something in the Ubuntu Docker image. Digging...

**Comment by avanier** [comment]
Ok, I've been digging through all the Ubuntu Docker image tags on Xenial and I have yet to find something that works. So now my guess is that this gets broken by either a `pip` upgrade I'm not seeing, or an `apt-get` update.

Will keep investigating.

**Comment by avanier** [comment]
Ok, it seems the breakage can be resolved by bumping `python-ldap` to `3.1.0`, so I'm going to guess that the compile code of the python bindings expect something from the `ldap-dev` package that's not provided anymore, and whichever version of the Docker image we use has that upgraded when we install the packages.

Working to figure out a fix.

**Comment by dwang159** [comment]
Our docs are pretty unclear here. I've opened #491 to address this issue; feel free to leave feedback if the instructions added there are unclear/lacking.
In the end, I did just bump python-ldap to 3.1.0 and add some documentation for setting up MySQL. We'll be moving to Python 3 sometime later this quarter (since Python 2 EOL is coming up), so I don't think it makes sense to stay on python-ldap 2.4.9 for much longer.
